superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx

```diff
@@ -117,9 +117,9 @@ export default function PluginFilterSelect(props: PluginFilterSelectProps) {
   const labelFormatter = useMemo(
     () =>
       getDataRecordFormatter({
-        timeFormatter: finestTemporalGrainFormatter(data.map(el => el.col)),
+        timeFormatter: finestTemporalGrainFormatter(data.map(el => el[col])),
```

Comment:
I wonder if col may sometimes be an object if it's an adhoc column? If so, we may need to do `getColumnLabel(col)` on it first..